vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -813,6 +825,16 @@ void AdsImpl::OnAdNotificationClicked(const AdNotificationInfo& ad) {
 
   epsilon_greedy_bandit_processor_->Process(
       {ad.segment, mojom::AdNotificationEventType::kClicked});
+
+  auto ad_notification_clicked_covariate_log_entry =
+      std::make_unique<AdNotificationClickedCovariateLogEntry>();
+  ad_notification_clicked_covariate_log_entry->SetClicked(true);
+  CovariateLogs::Get()->SetCovariateLogEntry(
+      std::move(ad_notification_clicked_covariate_log_entry));
+
+  mojom::TrainingInstancePtr instance =
+      CovariateLogs::Get()->GetTrainingInstance();
+  AdsClientHelper::Get()->LogTrainingInstance(std::move(instance));
```

Comment:
Can we decouple this code to a util, something along the lines of `ad_notification_clicked_covariate_log_traning_util`

Reply:
Summarised in in last comment, closing this one